patch_creation_wizard

```diff
@@ -145,28 +145,51 @@ def get_md5sum_command(file_path):
 def generate_before_patch_md5sums(path_to_repo, diff_files):
     """Generate MD5sums for each file before patch application."""
     for diff_file, details in diff_files.items():
-        # Reverse the diff to see what the file looked like before
-        subprocess.check_call(
-            [
-                'patch', '-R', details['path_in_repo'],
-                os.path.join(os.getcwd(), PATCH_DIFF_PATH, diff_file),
-            ],
-            cwd=path_to_repo,
-        )
-        details['before_md5sum'] = subprocess.check_output(
-            get_md5sum_command(details['path_in_repo']),
-            cwd=path_to_repo,
-            shell=True,
-            executable='/bin/bash',
-        ).split(' ')[0]
-        # Now restore the file to what it should be
-        subprocess.check_call(
-            [
-                'patch', details['path_in_repo'],
-                os.path.join(os.getcwd(), PATCH_DIFF_PATH, diff_file),
-            ],
-            cwd=path_to_repo,
-        )
+        try:
```

Comment:
Maybe split this try and the next one to functions for better readability

Reply:
Hmm... they are part of the same 'action' so I feel like different functions is overkill, but do you think a newline would be sufficient separation between the components?

Reply:
Sounds legit